### PR TITLE
Better detction and reporting of image builds

### PR DESCRIPTION
**Pull request by adammhaile**
Added --workingdir option

### CODE REVIEW

**Comment by adammhaile** [comment]
@jupierce Go easy on me... it's my first Groovy code :P

File: jobs/build/ocp/Jenkinsfile [hunk 1]
```diff
@@ -401,8 +424,85 @@ node(TARGET_NODE) {
...
+                    "openshift-enterprise-docker-builder-docker",
+                    "openshift-enterprise-dockerregistry-docker",
+                    "openshift-enterprise-egress-router-docker",
+                    "openshift-enterprise-federation-docker",
+                    "openshift-enterprise-haproxy-router-docker",
+                    "openshift-enterprise-keepalived-ipfailover-docker",
+                    "openshift-enterprise-mediawiki",
+                    "openshift-enterprise-mediawiki-docker",
+                    "openshift-enterprise-node-docker",
+                    "openshift-e",
```
Comment (stevekuznetsov) [inline jobs/build/ocp/Jenkinsfile]: typo?

File: jobs/build/ocp/Jenkinsfile [hunk 2]
```diff
@@ -401,8 +424,85 @@ node(TARGET_NODE) {
...
             stage( "build images" ) {
-                // TODO: Create a dynamic .repo file pointing to the exact puddle we built instead of "building" so that we can run X.Y builds in parallel
-                sh "ose_images.sh --user ocp-build build_container --branch rhaos-${BUILD_VERSION}-rhel-7 --group base --repo http://download.lab.bos.redhat.com/rcm-guest/puddles/RHAOS/repos/aos-unsigned-building.repo"
+                dir( temp=true ) {
+                  // TODO: Create a dynamic .repo file pointing to the exact puddle we built instead of "building" so that we can run X.Y builds in parallel
+                  PUDDLE_REPO = "http://download.lab.bos.redhat.com/rcm-guest/puddles/RHAOS/repos/aos-unsigned-building.repo"
+                  def build_result = sh (
+                    script: "ose_images.sh --user ocp-build build_container --branch rhaos-${BUILD_VERSION}-rhel-7 --workingdir ${pwd()} --group base --repo ${PUDDLE_REPO}"
+                    returnStatus: true
+                  ) > 0
```
Comment (stevekuznetsov) [inline jobs/build/ocp/Jenkinsfile]: why `> 0` if you treat this as an integer later?

File: jobs/build/ocp/Jenkinsfile [hunk 3]
```diff
@@ -401,8 +424,85 @@ node(TARGET_NODE) {
...
+                    }
+                    if(IMG_FAILURES.size() > 0){
+                      fails = []
+                      for(String img: IMG_FAILURES){
+                        if(required_images.contains(img)){
+                          fails.add(img)
+                        }
+                      }
+                      if(fails.size() > 0){
+                        msg = "One or more required images failed to build.\n"
```
Comment (stevekuznetsov) [inline jobs/build/ocp/Jenkinsfile]: Why hide non-required image failures?

File: jobs/build/ocp/Jenkinsfile (hunk 1 above)
Comment (adammhaile) [inline jobs/build/ocp/Jenkinsfile]: yup... pushing fix

File: jobs/build/ocp/Jenkinsfile (hunk 2 above)
Comment (adammhaile) [inline jobs/build/ocp/Jenkinsfile]: No good reason. Fixed.

File: jobs/build/ocp/Jenkinsfile (hunk 3 above)
Comment (adammhaile) [inline jobs/build/ocp/Jenkinsfile]: Ask @jupierce

Comment (adammhaile) [inline jobs/build/ocp/Jenkinsfile]: To clarify, it now sends an email saying it's a partial build if there are any failures. But @jupierce decided that it was still useful to have pushed builds even if not all images completed.

Comment (stevekuznetsov) [inline jobs/build/ocp/Jenkinsfile]: Right, my comment was just that you have the full list of failures but are only exposing the ones that are required -- that may be annoying for someone debugging a failure

Comment (adammhaile) [inline jobs/build/ocp/Jenkinsfile]: Fixed... now I always list everything.

File: jobs/build/ocp/Jenkinsfile [hunk 4]
```diff
@@ -81,6 +101,9 @@ ${OSE_CHANGELOG}
 
 ===OpenShift Ansible changelog snippet===
 ${OA_CHANGELOG}
+
+${img_fail_msg}
```
Comment (jupierce) [inline jobs/build/ocp/Jenkinsfile]: Let's move this information up above the changelogs. Particularly information about failing images needs to be very visible.

File: jobs/build/ocp/Jenkinsfile [hunk 5]
```diff
@@ -401,8 +424,84 @@ node(TARGET_NODE) {
...
+                  }
+
+                  def success_list_file = "${pwd()}/logs/success_image_list"
+                  IMG_SUCCESSES = []
+                  if(fileExists(success_list_file)){
+                    def success_image_list = readFile(success_list_file)
+                    IMG_SUCCESSES = success_image_list.split("\\r?\\n");
+                  }
+
+                  required_images = [
```
Comment (jupierce) [inline jobs/build/ocp/Jenkinsfile]: Let's invert this logic and turn it into a acceptable_to_fail list. The logging images are the ones that are most problematic,

File: build-scripts/ose_images/ose_images.sh [hunk 6]
```diff
@@ -136,7 +136,7 @@ add_group_to_list() {
         add_to_list logging-curator-docker
         add_to_list metrics-deployer-docker
         if [ ${MAJOR_RELEASE} != "3.3" ] && [ ${MAJOR_RELEASE} != "3.4" ]  && [ ${MAJOR_RELEASE} != "3.5" ] ; then
-          # add_to_list logging-auth-proxy-docker  
+          # add_to_list logging-auth-proxy-docker
```
Comment (jupierce) [inline build-scripts/ose_images/ose_images.sh]: We should be able to turn these back on with your PR.

**Comment by adammhaile** [comment]
@jupierce added puddle_status.sh
Was unsure if you wanted those changes on OpenShiftOnline as well as it didn't have the early/latest hack. Let me know if you want it there too. Easy to add.

**Pull request closed without merging**